spec/lib/pact/provider_verifier/app_spec.rb

```diff
@@ -56,6 +57,60 @@ module ProviderVerifier
             subject
           end
         end
+
+        context "when fail_if_no_pacts_found is false" do
+          before do
+            allow(options).to receive(:fail_if_no_pacts_found).and_return(false)
+          end
+
+          context "when no pacts are found" do
+            before do
+              allow(AggregatePactConfigs).to receive(:call).and_return([])
+            end
+
+            it { is_expected.to be true }
+          end
+        end
+
+        context "when fail_if_no_pacts_found is true" do
+          before do
+            allow(options).to receive(:fail_if_no_pacts_found).and_return(true)
+          end
+
+          context "when no pacts are found" do
+            before do
+              allow(AggregatePactConfigs).to receive(:call).and_return([])
+            end
+
+            it { is_expected.to be false }
+          end
+
+          context "when pacts are found and successfully verified" do
+            before do
+              allow(AggregatePactConfigs).to receive(:call).and_return([{}])
+              allow(Cli::RunPactVerification).to receive(:call).and_return(0)
+            end
+
+            it { is_expected.to be true }
+          end
+        end
+
+        context "multiple pacts need to be verified" do
+          before do
+            allow(AggregatePactConfigs).to receive(:call).and_return([{}, {}])
+          end
+
+          context "at least one pact fails verification" do
```

Comment:
Realized that my initial implementation broke this functionality. Seems pretty important, so added this spec